nservicebus/scheduling-with-nservicebus.md

```diff
@@ -5,19 +5,19 @@ tags:
 - Scheduler
 ---
 
-With the NServiceBus scheduling capability you can schedule a task to be executed repeatedly in a given interval.
+The NServiceBus Scheduler is a lightweight/non-durable API that helps to schedule a task that needs to be executed repeatedly based on a specified interval. The scheduling infrastructure leverages the reliable messaging approach and the NServiceBus core functionality. This allows scheduling to include features such as built in retries and forwarding to the error queue. 
 
-## Use cases
+## When not to use it
 
-You could use the scheduler to repeatedly poll some folder on your HDD for a file, schedule a reoccurring re-index of a database in a multi-tenant environment, and so on.
+- As soon as your task starts to get some branching logic (`if` or `switch` statements) or you start to add business logic, instead of a simple Bus.Send() or a Bus.SendLocal(), you should consider moving to a [saga](sagas-in-nservicebus.md) .
 
-Do not add any business logic in your tasks; instead, use [sagas](sagas-in-nservicebus.md).
+- Often times, rather than poll for a certain state using the Scheduler API, you can use an eventing model instead, where an event is published when that state transition occurs and take the necessary action in that event message handler.
 
-### Push versus Poll
+## How the scheduler works
 
-In the scenario where you are "polling" for some change in information state you should instead consider a "push" approach.
+The scheduler holds a all list of tasks scheduled in a non-durable in-memory dictionary. In version 3 and 4 tasks are scoped per `AppDomain`. In version 5 they are scoped per Bus instance.
 
-The scheduling API is often used to poll state, for example the file system or an external webservice. In these scenarios it is much more efficient to adopt a push based model. In the case of the file system you should instead consider a [FileSystemWatcher](http://msdn.microsoft.com/en-us/library/system.io.filesystemwatcher.aspx). In the case a webservice should instead consider a [WebHook](http://en.wikipedia.org/wiki/Webhook).
+When a new scheduled task is created it is given an unique identifier and stored in the endpoint's in-memory dictionary. The identifier for the task is sent in a message to the Timeout Manager for the message to be deferred for the specified time interval. When the specified time elapses, the Timeouts dispatcher returns the message containing the identifier to the endpoint with the scheduled task identifier, the endpoint uses that identifier to fetch and invoke the task from its internal list of tasks and executes them.

```

Comment:
is the Saxon genitive form correct? "endpoint's in-memory" shouldn't is simply be "endpoint in-memory"